src/FffGcodeWriter.cpp

```diff
@@ -1211,6 +1211,356 @@ void FffGcodeWriter::processSpiralizedWall(const SliceDataStorage& storage, Laye
     gcode_layer.spiralizeWallSlice(&mesh_config.inset0_config, wall_outline, last_wall_outline, seam_vertex_idx, last_seam_vertex_idx);
 }
 
+static bool polysIntersect(const Polygons& poly_a, const Polygons &poly_b)
+{
+    // only do the full intersection when the polys' BBs overlap
+    AABB bba(poly_a);
+    AABB bbb(poly_b);
+    return bba.hit(bbb) && poly_a.intersection(poly_b).size() > 0;
+}
+
+static bool polyOutlinesAdjacent(const ConstPolygonRef inner_poly, const ConstPolygonRef outer_poly, const coord_t max_gap)
+{
+    const coord_t max_gap2 = max_gap * max_gap;
+    const unsigned outer_poly_size = outer_poly.size();
+    for (unsigned line_index = 0; line_index < outer_poly_size; ++line_index)
+    {
+        const Point lp0 = outer_poly[line_index];
+        const Point lp1 = outer_poly[(line_index + 1) % outer_poly_size];
+        for (Point inner_poly_point : inner_poly)
+        {
+            if (LinearAlg2D::getDist2FromLineSegment(lp0, inner_poly_point, lp1) < max_gap2)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+static int findAdjacentEnclosingPoly(const ConstPolygonRef& enclosed_inset, const std::vector<ConstPolygonRef>& possible_enclosing_polys, const coord_t max_gap)
+{
+    // given an inset, search a collection of insets for the adjacent enclosing inset
+    Polygons enclosed;
+    enclosed.add(enclosed_inset);
+    for (unsigned enclosing_poly_idx = 0; enclosing_poly_idx < possible_enclosing_polys.size(); ++enclosing_poly_idx)
+    {
+        Polygons enclosing;
+        enclosing.add(possible_enclosing_polys[enclosing_poly_idx]);
+        // as holes don't overlap, if the insets intersect, it is safe to assume that the enclosed inset is inside the enclosing inset
+        if (polysIntersect(enclosing, enclosed) && polyOutlinesAdjacent(enclosed_inset, enclosing[0], max_gap))
+        {
+            return enclosing_poly_idx;
+        }
+    }
+    return -1;
+}
+
+static int findAdjacentPoly(const ConstPolygonRef& inset, const std::vector<ConstPolygonRef>& possible_adjacent_polys, const coord_t max_gap)
+{
+    // given an inset, search a collection of insets for an adjacent inset
+    for (unsigned poly_idx = 0; poly_idx < possible_adjacent_polys.size(); ++poly_idx)
+    {
+        if (polyOutlinesAdjacent(inset, possible_adjacent_polys[poly_idx], max_gap))
+        {
+            return poly_idx;
+        }
+    }
+    return -1;
+}
+
+static void processInsetsWithOptimizedOrdering(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage* mesh, const PathConfigStorage::MeshPathConfigs& mesh_config, const SliceLayerPart& part, unsigned int layer_nr, EZSeamType z_seam_type, Point z_seam_pos)
+{
+    const coord_t wall_line_width_0 = mesh->getSettingInMicrons("wall_line_width_0");
```

Comment:
You can obtain these settings from the config. Then these settings don't have to be parsed over again.

Moreover, some settings might be different on the initial layer, so getting the settings from the GCodePathConfig probably avoids some bugs.